### PR TITLE
Feature/milestone 4

### PR DESCRIPTION
**Pull request by HillaryOkello**
In this branch, we enhanced the Redux store by adding a books filter, implemented the categoryfilter component and kept the filter value in the Redux store.

### CODE REVIEW

File: src/reducers/filter.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { CHANGE_FILTER } from '../actions';
+
+const filterReducer = (state = {}, action) => {
```
Comment (rindrajosia) [inline src/reducers/filter.js]: - [ ] [REQUIRED] the default state of this reducer should be  'All' [screenshot from Pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone4/ItDefaultsToAll.png)

File: src/containers/BooksList.js [hunk 2]
```diff
@@ -3,16 +3,26 @@ import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 import Book from '../components/Book';
 import * as actions from '../actions';
+import CategoryFilter from '../components/categoryFilter';
 
 function BooksList(props) {
```
Comment (rindrajosia) [inline src/containers/BooksList.js:8]: - [ ] [REQUIRED] it implements `handleFilterChange` and passes it down to `CategoryFilter` component [screenshot from pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone4/HandelChangeFilter.png)

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Hi, 

Good job so far!
There are some issues to revise before continuing to the next project but hang in there and read the comments below with care. You can do it! :muscle:


## Required Changes ♻️

- [ ] I had an error related to proTypes when I open your APP on my local. It will be nice if you fix it:


- [ ] In order to make your readme more descriptive, please add the description of your project.


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Please consider:

- [ ] To add a screenshot in your readme file.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/containers/BooksList.js [hunk 3]
```diff
@@ -27,18 +37,23 @@ function BooksList(props) {
           </tr>
         </thead>
         <tbody>
-          {displayBook}
+          {displayBook()}
         </tbody>
       </table>
+      <CategoryFilter />
```
Comment (bellom) [inline src/containers/BooksList.js]: @HillaryOkello @kunmi02,

I think we missed this 👇  guys, the previous reviewer also mentioned it, up there 👆 

image


So we need to implement `handleFilterChange` function and pass it down to `CategoryFilter` component.

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @HillaryOkello @kunmi02 :clap: 

**Great work making `almost` all the necessary corrections from the previous review.**

# Status: Changes Required! ♻️ 

**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Review by oloomoses — APPROVED** [review]
Hi team, 

Your milestone is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_Slack: Oloo Moses_

**Pull request merged**